internal/controllers/clusterextension_controller.go

```diff
@@ -184,8 +184,8 @@ func (r *ClusterExtensionReconciler) reconcile(ctx context.Context, ext *ocv1alp
 		//  This is not ideal, and we should consider a more nuanced approach that resolves
 		//  as much status as possible before returning, or at least keeps previous state if
 		//  it is properly labeled with its observed generation.
-		ext.Status.ResolvedBundle = nil
-		ext.Status.InstalledBundle = nil
+		ext.Status.Resolution.Bundle = nil
+		ext.Status.Install.Bundle = nil
```

Comment:
Anywhere this change has been made runs a risk of accessing a `nil` pointer for the `ext.Status.Resolution` and `ext.Status.Install` fields. It looks like this is why some of the tests are failing (although maybe not caused by this line explicitly).

All areas where we update to this should perform some kind of nil check, and if the parent field is nil, set it. Something like:
```suggestion
		if ext.Status.Resolution == nil {
		        ext.Status.Resolution = &ocv1alpha1.ResolutionStatus{}
		}
		ext.Status.Resolution.Bundle = nil
	        if ext.Status.Install == nil {
		        ext.Status.Install = &ocv1alpha1.InstallStatus{}
		}
		ext.Status.Install.Bundle = nil
```

Since you'd have to likely do this `nil` check on every place we set this, it might be helpful to extract it into a helper function for each status that looks something like:
```go
func setResolutionStatus(ext *ocv1alpha1.ClusterExtension, resStatus *ocv1alpha1.ResolutionStatus) { ... }

func setInstallStatus(ext *ocv1alpha1.ClusterExtension, installStatus *ocv1alpha1.InstallStatus) { ... }
```